pkgs/development/python-modules/tyro/default.nix

```diff
@@ -0,0 +1,77 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  pytestCheckHook,
+  hatch,
+  typeguard,
+  docstring-parser,
+  typing-extensions,
+  rich,
+  shtab,
+  # Test dependencies
+  pyyaml,
+  pytest,
+  pytest-cov,
+  pytest-xdist,
+  omegaconf,
+  attrs,
+  torch,
+  pyright,
+  ruff,
+  mypy,
+  numpy,
+  flax,
+  pydantic,
+  coverage,
+  eval-type-backport,
+  jax,
+}:
+
+buildPythonPackage rec {
+  pname = "tyro";
+  version = "0.9.16";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "brentyi";
+    repo = "tyro";
+    tag = "v${version}";
+    hash = "sha256-dI6U9t4Y5TxhB710uuTmr9IgRLOEvZ9ZEWyHSWmrxag=";
+  };
+
+  build-system = [ hatch ];
+
+  dependencies = [
+    typeguard
+    docstring-parser
+    typing-extensions
+    rich
+    shtab
+  ];
+
+  nativeCheckInputs = [
+    omegaconf
+    torch
+    pydantic
+    jax
+    flax
+    attrs
+    pytestCheckHook
+  ];
+
+  checkPhase = ''
+    pytest
+  '';
+
+  doCheck = true;
```

Comment:
now that you have added the nativeCheckInputs you can remove checkPhase and doCheck attributes.